src/IlluminaSampleSheet/V2/HeaderSection.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace MLL\Utils\IlluminaSampleSheet\V2;
+
+use MLL\Utils\IlluminaSampleSheet\SectionInterface;
+
+class HeaderSection implements SectionInterface
+{
+    private const FILE_FORMAT_VERSION = '2';
+
+    private string $fileFormatVersion = self::FILE_FORMAT_VERSION;
+
+    private string $runName;
+
+    private ?string $runDescription = null;
+
+    private ?string $instrumentType = null;
+
+    private ?string $instrumentPlatform = null;
+
+    /** @var array<string, string> */
+    private array $customParams = [];
+
+    public function __construct(string $runName)
+    {
+        $this->runName = $runName;
+    }
+
+    public function setRunDescription(string $runDescription): HeaderSection
+    {
+        $this->runDescription = $runDescription;
+
+        return $this;
+    }
+
+    public function setInstrumentType(string $instrumentType): HeaderSection
+    {
+        $this->instrumentType = $instrumentType;
+
+        return $this;
+    }
+
+    public function setInstrumentPlatform(string $instrumentPlatform): HeaderSection
+    {
+        $this->instrumentPlatform = $instrumentPlatform;
+
+        return $this;
```

Comment:
unnecessary

Reply:
IMHO this is very opinionated. The fluent interface pattern IHMO is more concise and easy-to-read.
What exactly are the drawbacks that makes it "unnecessary" for you?